railties/lib/rails/application/configuration.rb

```diff
@@ -88,11 +88,16 @@ def initialize(*)
       # for the default values associated with a particular version.
       def load_defaults(target_version)
         # To introduce a change in behavior, follow these steps:
-        # 1. Add an accessor on the target object (e.g. the ActiveJob class for global Active Job config).
-        # 2. Set a default value there preserving existing behavior for existing applications.
+        # 1. Add an accessor on the target object (e.g. the ActiveJob class for
+        #    global Active Job config).
+        # 2. Set a default value there preserving existing behavior for existing
+        #    applications.
         # 3. Implement the behavior change based on the config value.
-        # 4. In the section below corresponding to the next release of Rails, set the new value.
-        # 5. Add a commented out section in the `new_framework_defaults` template, setting the new value.
+        # 4. In the section below corresponding to the next release of Rails,
+        #    configure the default value. Initially, this should be set to what
+        #    existing apps expect (the old, compatible, value).
+        # 5. Add a commented out section in the `new_framework_defaults`
+        #    template, configuring the desired default value for that version.
```

Comment:
I clarified these comments a bit.

I thought about setting the default value to `:sqlcommenter` in 7.1, but then I noticed this thread, from 7/27: https://github.com/rails/rails/pull/45618/files#r932901994.

This change clarifies that the default value should initially be set to what existing apps expect (the old, compatible, value), and not necessarily the "new" value.

Reply:
```suggestion
        #    configure the default value. 
        # 5. Add a commented out section in the `new_framework_defaults`
        #    template, configuring the desired default value for that version.
```

> Initially, it should be set to what existing apps expect

Is not quite correct here. The fact that the underlying configuration value `ActiveRecord.query_logs_tag_format` is set to `:legacy` means apps NOT using `load_defaults 7.1` will have the existing value. So apps that _do_ set `load_defaults 7.1` (and/or the `new_framework_defaults` for 7.1) should be setting it to the new value, `:sqlcommenter`

Reply:
Ah, that makes sense. Glad to hear that we can update the default value in 7.1, instead of waiting until 7.2 or later.